nexus/pkg/server/stream.go

```diff
@@ -117,20 +117,9 @@ func (s *Stream) GetRun() *service.RunRecord {
 	return s.handler.GetRun()
 }
 
-// Close closes the stream's handler, writer, sender, and dispatcher.
-// This can be triggered by the client (force=False) or by the server (force=True).
-// We need the ExitRecord to initiate the shutdown procedure (which we
-// either get from the client, or generate ourselves if the server is shutting us down).
-// This will trigger the defer state machines (SM) in the stream's components:
-//   - when the sender's SM gets to the final state, it will Close the handler
-//   - this will trigger the handler to Close the writer
-//   - this will trigger the writer to Close the sender
-//
-// This will finish the Stream's wait group, which will allow the stream to be
-// garbage collected.
+// Gracefully wait for handler, writer, sender, dispatcher to shutdown cleanly
+// assumes an exit record has already been sent
 func (s *Stream) Close() {
-	// Close and wait for input channel to shutdown
-	close(s.inChan)
```

Comment:
im not sure we care to close it.
I do worry that we should be emptying it though, just to be good channel citizens

Here is a case:
1) connection C1 opens to stream X
2) connection C2 opens to stream X
3) both do stuff
4) connection C1 finishes the run at streamX
5) connection C2 is a bad citizen and sits around on the connection

Do we wait for C2 to close?  we could, but we dont strictly need to.
If we did this cleanly, it would likely make improper usage more obvious rather than our current behavior of closing the connection on them.   But at the same point, improper usage could hang a run, which could be even more annoying.

So my proposal at this stage is to prioritize the work that are planning on processing and dropping anything else that comes outside of expected operation.

Reply:
so i think there was a semantic i saw here that drains the channel before closing it, not sure if this is a pattern we want but here is a reference: https://www.leolara.me/blog/closing_a_go_channel_written_by_several_goroutines/

Reply:
there are lots of good nuggets there.   looks useful to incorporate some ideas.